third_party/coro/coro.c

```diff
@@ -324,7 +330,11 @@ trampoline (int sig)
          ".cfi_offset 30, -16\n"
          "\tmov x0, x20\n"
          "\tblr x19\n"
+# ifdef __clang__
```

Comment:
Here we are: my concerns regarding mangled/unmangled symbols on Linux/ARM64 confirmed. I tried to build Tarantool via clang-9 on odroid host and faced the following error:
```
$ uname -mrs
Linux 4.9.213-67 aarch64
$ apt-get -y install git build-essential cmake make zlib1g-dev libreadline-dev libncurses5-dev libssl-dev libunwind-dev libicu-dev clang-9
<snipped>
$ git clone https://github.com/tarantool/tarantool.git
<snipped>
# There are some issues with libunwind symbol resolving, so I disabled backtraces support for now.
$ CC=clang-9 CXX=clang++-9 cmake . -DENABLE_BACKTRACE=OFF
<snipped>
$ make
<snipped>
[ 63%] Linking CXX executable tarantool
../libcoro.a(coro.c.o): In function `coro_startup':
coro.c:(.text+0x78): undefined reference to `_abort'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
src/CMakeFiles/tarantool.dir/build.make:233: recipe for target 'src/tarantool' failed
make[2]: *** [src/tarantool] Error 1
CMakeFiles/Makefile2:3614: recipe for target 'src/CMakeFiles/tarantool.dir/all' failed
make[1]: *** [src/CMakeFiles/tarantool.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2
```

Hence, despite the clang toolchaing is chosen, the symbols to be resolved are unmangled. When I change `__clang__` to `__APPLE__`, everything builds fine. However, I'm not sure, whether this particular define is the right one to be checked here.

Reply:
Well, I see. Thanks for investigation! It would be nice to leave this link right in the commit message, but this is not obligatory.

Reply:
The link above mentioned (although, talking about Windows) that leading underscore is for ia32 only, not 64bit. Apparently, it is true for all targets under MacOS - we just need a better link here. 
Also, I see it works for x64 build just because we don't need a CORO_STARTUP - hence no call to _abort and _coro_startup.